pex/resolve/lock_resolver.py

```diff
@@ -103,9 +103,13 @@ def __init__(
         self._network_configuration = network_configuration
         self._password_entries = password_entries
         self._cache = cache
+
+        # Since a VCSArtifactDownloadManager is only used for VCS requirements, a build is both
+        # required and preferred by the user.
         self._build_configuration = attr.evolve(
-            build_configuration, allow_wheels=False, allow_builds=True, prefer_older_binary=False
+            build_configuration, allow_builds=True, prefer_older_binary=False
```

Comment:
I was initially confused why changes to `VCSArtifactDownloadManager` would improve the behaviour for what was tested. That test seems to only use PyPI requirements, not VCS ones.

My understanding is that this error happens eagerly, on construction of this `BuildConfiguration` object, and so when a `VCSArtifactDownloadManager` is constructed, it'll eagerly do the checking and throw the error here.

Is that correct?

Reply:
Yes.